lib/tool_shed/util/hg_util.py

```diff
@@ -395,21 +396,21 @@ def init_repository(repo_path):
     try:
         subprocess.check_output(['hg', 'init'], stderr=subprocess.STDOUT, cwd=repo_path)
     except Exception as e:
-        error_message = 'Error initializing repository: %s' % e
+        error_message = 'Error initializing repository: %s' % unicodify(e)
         if isinstance(e, subprocess.CalledProcessError):
-            error_message += "\nOutput was:\n%s" % e.output
+            error_message += "\nOutput was:\n%s" % unicodify(e.output)
         raise Exception(error_message)
 
 
 def changeset2rev(repo_path, changeset_revision):
     """
-    Return the revision number corresponding to a specified changeset revision.
+    Return the numeric revision number (as a string) corresponding to a specified changeset revision.
     """
     try:
         rev = subprocess.check_output(['hg', 'id', '-r', changeset_revision, '-n'], stderr=subprocess.STDOUT, cwd=repo_path)
     except Exception as e:
-        error_message = "Error looking for changeset '%s': %s" % (changeset_revision, e)
+        error_message = "Error looking for changeset '%s': %s" % (changeset_revision, unicodify(e))
         if isinstance(e, subprocess.CalledProcessError):
-            error_message += "\nOutput was:\n%s" % e.output
+            error_message += "\nOutput was:\n%s" % unicodify(e.output)
         raise Exception(error_message)
-    return int(rev.strip())
+    return rev.strip()
```

Comment:
What was the issue with this? I think I prefer the way it was, with casting to `str` by the caller when needed, especially since:
- `subprocess.check_output()` return a bytestring
- casting to int here finds any issue as early as possible.

Reply:
There's only one place in the code where we need `ctx_rev` to be numeric (RepositoryMetadata.numeric_revions), while everywhere else it should be a string.
I agree though that it's nice to check this is an integer,
so how about `return str(int(rev.strip())))` ?

Reply:
`changeset2rev` is called only in 3 places, so it's not a big problem to cast in the caller if needed. Anyway, I think I've fixed the toolshed tests by casting to `str` where needed in an updated commit, let me know if you dislike it!

Reply:
I would prefer it if ctx_rev was always a string, so I did that in my last commit. Does that seem like a reasonable compromise ?